docs/root/intro/arch_overview/http/upgrades.rst

```diff
@@ -14,7 +14,10 @@ pass through the default HTTP filter chain. To avoid the use of HTTP-only filter
 one can set up custom
 :ref:`filters <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.UpgradeConfig.filters>`
 for the given upgrade type, up to and including only using the router filter to send the HTTP
-data upstream.
+data upstream. Note that buffering is generally not compatible with upgrades, so if the
+:ref:`Buffer filter <envoy_v3_api_msg_extensions.filters.http.buffer.v3.Buffer>` is configured in
+the default HTTP filter chain it'll have to be excluded from the list of
```

Comment:
optional nit:
it'll have to be  -> it should probably be

Reply:
ah yes